pylint/checkers/unicode.py

```diff
@@ -0,0 +1,670 @@
+
+    Default to UTF-8 if no BOM is found
+
+    Args:
+        first_line: the first line of a file
+
+    Returns:
+        a codec name
+    """
+    for bom, codec in get_bom_sorted().items():
+        if first_line.startswith(bom):
+            return codec
+
+    return None
+
+
+class InvalidUnicodeModeChecker(BaseChecker):
+    """
+    Check that UTF-16 and UTF-32 are not used to encode Python Files.
+
+    Currently Python supports only UTF-8. See
+    https://stackoverflow.com/questions/69897842/ and https://bugs.python.org/issue1503789
+    for background.
+    This simplifies the BiDirectional Unicode check below as we only need to check
+    for UTF-8 bytes.
+
+    As this is a security related check we want to ensure that even once Python
+    support UTF-16 and UTF-32 it is not possible for an attacker to bypass our checks
+    by simply using a new encoding.
+    """
+
+    __implements__ = interfaces.IRawChecker
+
+    msgs = {
+        "E1800": (
+            "Use only UTF-8, not UTF-16 or UTF-32!",
+            "invalid-unicode-codec",
+            (
+                "UTF-16 / UTF-32 is currently not working properly with Python, "
+                "see https://stackoverflow.com/questions/69897842/"
+            ),
+        )
+    }
+    name = "invalid_unicode_encoding_checker"
+
+    @staticmethod
+    def is_invalid_codec(codec: Optional[str]) -> bool:
+        if codec is None:
+            return False
+        codec = codec.lower().replace("-", "").replace("_", "")
+        return codec.startswith("utf16") or codec.startswith("utf32")
+
+    def process_module(self, node: nodes.Module) -> None:
+        """
+        Perform the the actual checking by checking module stream
+
+        the module's content is accessible via node.stream() function
+        """
+        codec: Optional[str]
+        lines: Sequence[bytes]
+        invalid_codec_line: Optional[int] = None
+
+        # First try to detect encoding with PEP 263, won't work
+        # see https://bugs.python.org/issue1503789
+        # TO DO we could manually modify the stream to convert UTF16/UTF32 to ASCII
+        #      but why ;-) ?
+        try:
+            with node.stream() as stream:
+                codec, lines = detect_encoding(stream.readline)
+        except SyntaxError:
+            pass
+        else:
+            if self.is_invalid_codec(codec):
+                invalid_codec_line = len(lines)
+
+        # If there is no invalid codec already, lets try to determine a BOM,
+        # even so Python currently doesn't like non UTF-8 BOMs ....
+        if invalid_codec_line is None:
+            with node.stream() as stream:
+                codec = extract_codec_from_bom(stream.readline())
+            if self.is_invalid_codec(codec):
+                invalid_codec_line = 1
+
+        if invalid_codec_line is not None:
+            self.add_message(
+                "invalid-unicode-codec",
+                line=invalid_codec_line,
```

Comment:
```suggestion
                node=node,
                line=invalid_codec_line,
```

Reply:
Same as below. The node is only the module, which is not helpful at all

Reply:
We're in the process of making `node` mandatory, so we'll need to add this at some point. You can still use `line` if you want to override `nodes.Module.lineno` which is `0` by default.